web/modules/weather_data/src/Service/WeatherDataService.php

```diff
@@ -657,6 +657,57 @@ public function getCurrentConditionsFromGrid($wfo, $gridX, $gridY)
         ];
     }
 
+    public function getHourlyPrecipitation(
+        $wfo,
+        $gridX,
+        $gridY,
+        $now = false,
+        $self = false,
+    ) {
+        date_default_timezone_set("America/New_York");
+
+        if (!$self) {
+            $self = $this;
+        }
+
+        $wfo = strtoupper($wfo);
+        if (!($now instanceof \DateTimeImmutable)) {
+            $now = new \DateTimeImmutable();
+        }
+
+        $place = $self->getPlaceFromGrid($wfo, $gridX, $gridY);
+        $timezone = $place->timezone;
+
+        $forecast = $self->getFromWeatherAPI("/gridpoints/$wfo/$gridX,$gridY")
+            ->properties;
+
+        $periods = [];
+
+        foreach ($forecast->quantitativePrecipitation->values as $quantPrecip) {
+            $valid = $quantPrecip->validTime;
+            $value = $quantPrecip->value;
+
+            $valid = explode("/", $valid);
+            $start = \DateTimeImmutable::createFromFormat(
+                \DateTimeInterface::ISO8601_EXPANDED,
+                $valid[0],
+            )->setTimeZone(new \DateTimeZone($timezone));
+
+            $duration = new \DateInterval($valid[1]);
+            $end = $start->add($duration);
+
+            if ($end >= $now) {
+                $periods[] = (object) [
+                    "start" => $start->format("g A"),
+                    "end" => $end->format("g A"),
+                    "value" => round($value * 0.03937008, 1),
```

Comment:
What's the constant we are using here?

Reply:
Good catch. This is the conversion from mm to inches. It should be factored into a constant and/or put into the unit conversion trait thingy. Will do that!